pkg/rhos/ocpgwdeployer.go

```diff
@@ -200,7 +203,15 @@ func (d *ocpGatewayDeployer) deployGWNode(gatewayCount int, groupName string,
 		gatewayNodesToDeploy := gatewayCount - numGatewayNodes
```

Comment:
In Line 159, we are calling the d.msDeployer.List() which lists the Machines which have the "submariner.io/gateway=true" label. However, I do not see that label added to the MachineSet Yaml[1]. Wont we run into issues (meaning more number of Gateway nodes getting spawned than what we desire) when the cloud-prepare APIs are invoked periodically while it takes time for the actual node to show up? 

[1] The MachineSet is not added with the "submariner.io/gateway=true", however the MachineSet.spec.template.spec.labels include the "submariner.io/gateway=true" label which is actually applied on the final Node that gets created as part of MachineSet.

Reply:
The API get the list by checking the label inside spec [1] and not the nodes. So this should exist when the machineset is created successfully. So we should be good right?

[1]https://github.com/submariner-io/cloud-prepare/blob/devel/pkg/ocp/machinesets.go#L205

Reply:
Oh yeah, that's a bit confusing unless we carefully look at where the labels are checked. Thanks for clarifying.